src/app/backend/userlinks/userlinks.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Kubernetes Dashboard Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userlinks
+
+import (
+	"encoding/json"
+	"log"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/kubernetes/dashboard/src/app/backend/api"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sClient "k8s.io/client-go/kubernetes"
+)
+
+const (
+	annotationObj     = "alpha.dashboard.kubernetes.io/links"
+	apiserverProxyURL = "http://{{apiserver-proxy-url}}"
```

Comment:
we should remove the 'http' prefix from the pattern. we do not know if http or https is used. 

comment: removing http might makes the pattern less recognizable as URL, but easier to handle.

The protocol prefix seems to be cut off in the code, anyway